dali/operators/reader/loader/tar_utils_test.cc

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/loader/tar_utils.h"
+#include <fcntl.h>
+#include <gtest/gtest.h>
+#include <libtar.h>
+#include <algorithm>
+#include <future>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "dali/operators/reader/loader/filesystem.h"
+#include "dali/util/std_file.h"
+
+namespace dali {
+namespace detail {
+
+TEST(LibTarUtilsTestSimple, Interface) {
+  std::string filepath(dali::filesystem::join_path(std::getenv("DALI_EXTRA_PATH"),
+                                                   "db/webdataset/MNIST/devel-1.tar"));
+  TarArchive archive(TarArchive(TarArchive(FileStream::Open(filepath, false, false))));
+  ASSERT_TRUE(archive.NextFile());
+  ASSERT_TRUE(archive.IsAtFile());
+  ASSERT_EQ(archive.GetFileName(), "0.jpg");
+  int filesize = archive.GetFileSize();
+  ASSERT_FALSE(archive.Eof());
+  archive.Read(10);
+  ASSERT_FALSE(archive.Eof());
+  ASSERT_EQ(archive.GetFileSize(), filesize);
+  archive.Read();
+  ASSERT_TRUE(archive.Eof());
+}
+
+TEST(LibTarUtilsTestSimple, LongNameIndexing) {
+  std::string filepath(dali::filesystem::join_path(std::getenv("DALI_EXTRA_PATH"),
+                                                   "db/webdataset/sample-tar/gnu.tar"));
+  TarArchive archive(TarArchive(TarArchive(FileStream::Open(filepath, false, false))));
+  std::string name_prefix(128, '#');
+  for (int idx = 0; idx < 1000; idx++) {
+    ASSERT_EQ(archive.GetFileName(), name_prefix + to_string(idx));
+    archive.NextFile();
+  }
+}
+
+void TestArchiveEntries(TarArchive& archive, const std::vector<std::string>& prefixes, int beg,
+                        int end, bool preread) {
+  for (int idx = beg; idx < end; idx++) {
+    for (int prefix_idx = 0; prefix_idx < static_cast<int>(prefixes.size()); prefix_idx++) {
```

Comment:
Is there any reason to not use `size_t` here? I don't like unsigned `size_t`, but you have two code uglifying casts (here and in the codition). Alternatively, you can do:
```suggestion
    for (int prefix_idx = 0, num_prefixes = prefixes.size(); prefix_idx < num_prefixes; prefix_idx++) {
```
and use num_prefixes - but in this instance I think that `size_t` is more practical.

Reply:
And what about `size_t`?

Reply:
No I mean that if I I have:
`for (int prefix_idx = 0; prefix_idx < prefixes.size(); prefix_idx++) `
then it throws warnings